File: src/patterns/structural/bridge/advancedremote.nim
# internal
import
    patterns/structural/bridge/[
        device,
        remote
    ]


type
    AdvancedRemote*[TDevice: Device] = ref object of Remote[TDevice]


proc newAdvancedRemote*[TDevice: Device](device: TDevice): AdvancedRemote[TDevice] =
    return AdvancedRemote[TDevice](device: device)

proc mute*(self: AdvancedRemote): bool =
    return self.device.setVolume(0)

File: src/patterns/structural/bridge/advancedremote.go
package bridge


type AdvancedRemote struct {
	*Remote
}


func NewAdvancedRemote(device Device) *AdvancedRemote {
	return &AdvancedRemote{
		&Remote{device: device},
	}
}

func (self *AdvancedRemote) Mute() bool {
	return self.device.SetVolume(0)
}